src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerCache.cs

```diff
@@ -85,7 +91,7 @@ public Entry GetCacheEntry(ControllerContext controllerContext)
             }
             else
             {
-                entry = new Entry(null, items, executor);
+                entry = new Entry(null, staticallyDefinedFilterItems, executor);
             }
```

Comment:
> but we need to call the providers every request.

The providers are getting called for every request via the `GetFiltersFromEntry` method.